rpc/serve.go

```diff
@@ -19,34 +19,24 @@ func Serve(lis net.Listener, bootstrapClient capnp.Client) error {
 		return err
 	}
 	// Accept incoming connections
+	defer bootstrapClient.Release()
 	for {
-		rwc, err := lis.Accept()
+		conn, err := lis.Accept()
 		if err != nil {
 			// Since we took ownership of the bootstrap client, release it after we're done.
-			if !bootstrapClient.IsValid() {
-				err = errors.New("the bootstrap client was already released")
-			}
 			bootstrapClient.Release()
 			return err
 		}
 
+		// the RPC connection takes ownership of the bootstrap interface and will release it when the connection
+		// exits, so use AddRef to avoid releasing the provided bootstrap client capability.
+		opts := Options{
+			BootstrapClient: bootstrapClient.AddRef(),
```

Comment:
Nit:  `bootstrapClient` is of type `capnp.Client`, so the `Client` suffix is redundant.  I suggest renaming to `boot`.

Reply:
I guess that is a style choice. Personally I prefer more descriptive names to point the casual reader in the right direction. If the style here is to keep things short and not repeat yourself then I'll remove it. Not sure what you guys prefer. (is there a style guide?)

Reply:
The general Go convention is to let the types do (some of) the talking.  The variable name `boot` tells us how the `capnp.Client` is to be used, so it's adding new information.  `bootstrapClient capnp.Client` kind of "stutters", and has the unfortunate effect of lengthening Go's already-quite-long lines.

All this to say that I have a moderate preference for the DRY approach.  But I don't want to die on this hill, either.

Reply:
I'll give it a try, thanks for explaining